attest/core/src/nonce.rs

```diff
@@ -2,36 +2,31 @@
 
 //! Nonce structures
 
-use crate::{error::NonceError, impl_sgx_newtype_for_bytestruct};
+use crate::error::NonceError;
 use alloc::{vec, vec::Vec};
 use core::{
     fmt::{Display, Formatter, Result as FmtResult},
     write,
 };
 use hex_fmt::HexFmt;
-use mc_sgx_types::sgx_quote_nonce_t;
+use mc_sgx_core_types::QuoteNonce;
 use mc_util_encodings::{Error as EncodingError, FromHex, ToHex};
 use rand_core::{CryptoRng, RngCore};
 use serde::{Deserialize, Serialize};
 use subtle::{Choice, ConstantTimeEq};
 
 /// A trait used to define common operations on nonce values
 pub trait Nonce:
-    AsRef<[u8]>
-    + PartialEq
-    + Sized
-    + for<'bytes> TryFrom<&'bytes [u8]>
-    + TryFrom<Vec<u8>>
-    + ConstantTimeEq
+    AsRef<[u8]> + PartialEq + Sized + for<'bytes> TryFrom<&'bytes [u8]> + ConstantTimeEq
```

Comment:
The `TryFrom<Vec<u8>>` was unecessary since it created a new vec anyway.
This is overall more ergonomic with  `TryFrom<&'bytes [u8]>`.